oxenc/bt_serialize.h

```diff
@@ -1097,6 +1119,35 @@ class bt_dict_consumer : private bt_list_consumer {
         next_dict(dict);
     }
 
+    /// This function combines the use of require() and consume_integer(). It is identical
+    /// to executing the following:
+    ///
+    ///     const char* key = "x";
+    ///     oxenc::bt_dict_consumer btdp{data};
+    ///
+    ///     btdp.require(key);
+    ///     int64_t x = btdp.consume_integer<int64_t>();
+    ///
+    template <typename IntType>
+    auto require_integer(std::string_view find) {
+        require(find);
+        return consume_integer<IntType>();
+    }
+
+    /// This function combines the use of require() and consume_integer(). It is identical
+    /// to executing the following:
+    ///
+    ///     const char* key = "x";
+    ///     oxenc::bt_dict_consumer btdp{data};
+    ///
+    ///     btdp.require(key);
+    ///     std::string x = btdp.consume_string();
+    ///
+    auto require_string(std::string_view find) {
+        require(find);
+        return consume_string();
+    }
```

Comment:
There needs to be more here: `require_string_view`, `require_list`, `require_dict_consumer`, and so on.  Basically there should be 1-1 mappings with all the `consume_...` functions.